Tests/test_image_draft.py

```diff
@@ -2,36 +2,73 @@
 
 from PIL import Image
 
-CODECS = dir(Image.core)
-FILENAME = "Tests/images/hopper.jpg"
-DATA = tostring(Image.open(FILENAME).resize((512, 512)), "JPEG")
-
-
-def draft(mode, size):
-    im = fromstring(DATA)
-    im.draft(mode, size)
-    return im
-
 
 class TestImageDraft(PillowTestCase):
-
     def setUp(self):
-        if "jpeg_encoder" not in CODECS or "jpeg_decoder" not in CODECS:
+        codecs = dir(Image.core)
+        if "jpeg_encoder" not in codecs or "jpeg_decoder" not in codecs:
             self.skipTest("jpeg support not available")
 
+    def draft_roundtrip(self, in_mode, in_size, req_mode, req_size):
+        im = Image.new(in_mode, in_size)
+        data = tostring(im, 'JPEG')
+        im = fromstring(data)
+        im.draft(req_mode, req_size)
+        return im
+
     def test_size(self):
-        self.assertEqual(draft("RGB", (512, 512)).size, (512, 512))
-        self.assertEqual(draft("RGB", (256, 256)).size, (256, 256))
-        self.assertEqual(draft("RGB", (128, 128)).size, (128, 128))
-        self.assertEqual(draft("RGB", (64, 64)).size, (64, 64))
-        self.assertEqual(draft("RGB", (32, 32)).size, (64, 64))
+        for in_size, req_size, out_size in [
+            ((435, 361), (2048, 2048), (435, 361)),  # bigger
+            ((435, 361), (435, 361), (435, 361)),  # same
+            ((128, 128), (64, 64), (64, 64)),
+            ((128, 128), (32, 32), (32, 32)),
+            ((128, 128), (16, 16), (16, 16)),
+
+            # large requested width
+            ((435, 361), (218, 128), (435, 361)),  # almost 2x
```

Comment:
Does this mean that if we request a draft image of (218, 128), we're getting something larger than that?